### PR TITLE
Add submitter for PAI platform in sqlflow.runtime

### LINKED ISSUE #2590: Supports submitting pai/alisa tasks in sqlflow.runtime
As the discussion of #2494 and #2562 , SQLFlow should be a compiler, which generates target code instead of executing the intermediate representation (IR).

We now have a Go function https://github.com/sql-machine-learning/sqlflow/blob/cafd7cc92724a6a162756ebef444832484d578cc/pkg/executor/pai.go#L214 that submits pai/alisa tasks. This prevents us from decoupling codegen and execution, so we have to implement a Python module/function in `sqlflow.runtime` that supports submitting pai/alisa tasks.

### PR DESCRIPTION
**Pull request by lhw362950217**
Fixes #2590 

This PR is a prototype of submitter for PY-TF, we only implemented the tensorflow train part, the predict, explain parts will be implemented in following PRs. Also, XGBoost and other framework will be in following PRs too.

### CODE REVIEW

File: python/sqlflow_submitter/pai/submitter.py [hunk 1]
```diff
@@ -0,0 +1,301 @@
...
+             "./sqlflow_models", "requirements.txt", TRAIN_PARAMS_FILE)
+
+
+def save_model_to_sqlfs(datasource, model_oss_path, model_name):
+    # (TODO: save model to sqlfs)
+    pass
+
+
+# (TODO: lhw) adapt this interface after we do feature derivation in Python
+def submit_pytf_train(datasource, estimator, select, validation_select,
```
Comment (shendiaomo) [inline python/sqlflow_submitter/pai/submitter.py]: I think we'll have too many combinations this way: 
- submit_pytf_train
- submit_pyxgb_train
- submit_pytf_predict
- ...

Please try to find a simpler design.

File: python/sqlflow_submitter/pai/submitter.py [hunk 2]
```diff
@@ -0,0 +1,301 @@
...
+                      model_params, model_name, pre_trained_model, **train_params):
+    """
+    This function submit PY-TF train task to PAI platform
+
+    Parameters
+    ----------
+    datasource: string
```
Comment (shendiaomo) [inline python/sqlflow_submitter/pai/submitter.py]: Use this style to format docstrings? https://github.com/google/tangent/blob/master/tangent/ast.py
This is also what TensorFlow's docstring style.

File: python/sqlflow_submitter/pai/cluster_conf.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+        "train.num_workers":   1,
+        "train.worker_cpu":    400,
+        "train.worker_gpu":    0,
+        "train.ps_cpu":        200,
+        "train.ps_gpu":        0,
+        "train.num_evaluator": 0,
+        "train.evaluator_cpu": 200,
+        "train.evaluator_gpu": 0,
+    }
+    for (k, v) in attrs.entries():
```
Comment (shendiaomo) [inline python/sqlflow_submitter/pai/cluster_conf.py]: 1. Does the Python `dict` has an `entries` method? It seems should be `items`.
2. We cannot `raise` a `str` in Python3, the object raised must be a instance of `Exception`
3. L27-30 should be an atomic operation. I think the following code is atomic and is readable:
```python
if not all(isinstance(v, int) for v in attr.values()):
    raise SQLFlowDiagnostic("xxx")
default_map.update(attr)
```

**Review by shendiaomo — COMMENTED** [review]
Please follow the style of https://github.com/google/tangent/blob/master/tangent/ast.py for each function to import readability

**Review by Yancey0623 — COMMENTED** [review]
How about adding some unit tests?

File: python/sqlflow_submitter/pai/submitter.py (hunk 1 above)
Comment (lhw362950217) [reply python/sqlflow_submitter/pai/submitter.py]: Whether or not add a unified-train, we always need these function. After a discussion with @Yancey1989 , I think we may keep them now and add a unified interface later.

File: python/sqlflow_submitter/pai/submitter.py (hunk 2 above)
Comment (lhw362950217) [reply python/sqlflow_submitter/pai/submitter.py]: done

File: python/sqlflow_submitter/pai/cluster_conf.py (hunk 3 above)
Comment (lhw362950217) [reply python/sqlflow_submitter/pai/cluster_conf.py]: done

**Comment by shendiaomo** [comment]
> Please follow the style of https://github.com/google/tangent/blob/master/tangent/ast.py for each function to import readability

BTW, we'd better take `codegen` into consideration when designing this module.

**Review by shendiaomo — APPROVED** [review]

File: python/runtime/pai/submitter.py [hunk 4]
```diff
@@ -0,0 +1,379 @@
...
+
+    # submit pai task to execute the training
+    cmd = get_pai_tf_cmd(conf, JOB_ARCHIVE_FILE, PARAMS_FILE,
+                         ENTRY_DIR + "tensorflow.py", model_name, path_to_save,
+                         train_table, val_table, "", project, cwd)
+    submit_pai_task(cmd, datasource)
+
+    # save trained model to sqlfs
+    save_model_to_sqlfs(datasource, path_to_save, model_name)
+    drop_tmp_tables([train_table, val_table], datasource)
```
Comment (typhoonzero) [inline python/runtime/pai/submitter.py:379]: We should call `drop_tmp_tables` when there's any exception or error occurs in above process.

**Pull request merged**